app/src/main/java/com/mealsmadeeasy/ui/splash/SplashActivity.kt

```diff
@@ -1,15 +1,122 @@
 package com.mealsmadeeasy.ui.splash
 
 import android.os.Bundle
+import android.content.Intent
+import android.support.v4.app.FragmentActivity
+import android.util.Log
 import com.mealsmadeeasy.ui.BaseActivity
 import com.mealsmadeeasy.ui.home.HomeActivity
+import com.mealsmadeeasy.R
+import android.view.View
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.android.gms.tasks.Task
+
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.GoogleAuthProvider
+import android.support.design.widget.Snackbar
+import android.widget.Button
+import com.google.android.gms.common.SignInButton
+import kotlinx.android.synthetic.main.activity_splash.*
+
+
+class SplashActivity : BaseActivity(), View.OnClickListener {
+
+    private val TAG = "SplashActivity"
+    private val RC_SIGN_IN = 9001
+
+    private var mAuth: FirebaseAuth? = null
+    private var mGoogleSignInClient: GoogleSignInClient? = null
```

Comment:
We can Kotlin-ize this better by making these look like `private lateinit var auth: FirebaseAuth` instead.